lib/replicator.js

```diff
@@ -2086,16 +2086,32 @@ module.exports = class Replicator {
     this._maybeResolveIfAvailableRanges()
   }
 
-  _closeSession (session) {
-    session.close().catch(safetyCatch)
+  _closeSession () {
+    this.core.active--
+
+    // we were the last active ref, so lets shut things down
+    if (this.core.active === 0 && this.core.sessions.length === 0) {
+      this.destroy()
+      this.core.close().catch(safetyCatch)
+      return
+    }
+
+    // in case one session is still alive but its been marked for auto close also kill it
+    if (this.core.sessions.length === 1 && this.core.active === 1 && this.core.sessions[0].autoClose) {
```

Comment:
2 final sanity checks:
- Can the case `(this.core.active === 1 && this.core.sessions.length === 0)` or `this.core.active === 0 && this.core.sessions.length === 1` ever happen? (we only consider 0-0 and 1-1)
- is it relevant to consider the case of multiple remaining sessions, all marked for auto close?

Reply:
only one autoclose one can exist (its a corestore option)

Reply:
Ya active can easily be 1 or more here with no explicit sessions. That just means more replication sessions exist which will then go through this code later